assets/news-api/components/EditAPIToken.jsx

```diff
@@ -0,0 +1,279 @@
+        const title = existing ?
+            gettext('Edit API Token') :
+            gettext('Create New API Token');
+
+        return (
+            <div className='tab-pane active company-api__token-edit' id='company-api-token'>
+                <form onSubmit={(event) => {event.preventDefault();}}>
+                    <div className="list-item__preview-form">
+                        <div className="card mt-3 d-block">
+                            <div className="card-header d-flex flex-row">{title}</div>
+                            <div className="card-body">
+                                <ExpiryDateInput
+                                    name="expiry"
+                                    label={gettext('Expires:')}
+                                    value={getDateInputDate(this.state.token.expiry)}
+                                    onChange={this.onExpiryChange}
+                                    error={get(this.state, 'errors.expiry')}
+                                />
+
+                                <CheckboxInput
+                                    name='is_enabled'
+                                    label={gettext('Enabled')}
+                                    value={this.state.token.enabled}
+                                    onChange={this.onEnabledChange}
+                                />
+                            </div>
+                            <div className="card-footer d-flex">
+                                <button
+                                    className="btn btn-outline-primary ml-auto"
+                                    onClick={this.saveToken}
+                                    disabled={Object.keys(this.state.errors).length > 0}
+                                >
+                                    {existing ? gettext('Save') : gettext('Generate Token')}
+                                </button>
+                                {existing && (
+                                    <button
+                                        className='btn btn-outline-secondary ml-3'
+                                        onClick={() => this.setState({action: EDITOR_ACTIONS.PREVIEW})}
+                                    >
+                                        {gettext('Cancel')}
+                                    </button>
+                                )}
+                            </div>
+                        </div>
+                    </div>
+                </form>
+            </div>
+        );
+    }
+
+    renderPreview() {
+        const expiry = moment(get(this.state, 'token.expiry'));
+        const token = get(this.state, 'token.token', '');
+        const expiresLabel = isInPast(get(this.state, 'token.expiry')) ?
+            gettext('Expired On:') :
+            gettext('Expires On:');
+        const title = !get(this.state, 'token.enabled', false) ?
+            gettext('API Token - Disabled') :
```

Comment:
Probably can make 'Disabled' be more significant as it is not catching attention. Use red color probably?